CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
```

Comment:
I agree that giving a reason of why we do so is a very good idea. But I also agree we should remain 100% transparent.

Proposition:

The Scala.js project has a strict policy regarding coding style. It is our firm belief that this is one of the cornerstones to maintain a clean, consistent and maintainable codebase.

This document tries to document the style in use as much as possible to make it easier for everyone to contribute. (<-- this one is true, this is the reason we write the document. Internally we never needed it, but now we do, because of the contributions we get)